core/src/validator.rs

```diff
@@ -1293,23 +1299,23 @@ fn wait_for_supermajority(
     cluster_info: &ClusterInfo,
     rpc_override_health_check: Arc<AtomicBool>,
     start_progress: &Arc<RwLock<ValidatorStartProgress>>,
-) -> bool {
+) -> (bool, bool) {
     if let Some(wait_for_supermajority) = config.wait_for_supermajority {
         match wait_for_supermajority.cmp(&bank.slot()) {
-            std::cmp::Ordering::Less => return false,
+            std::cmp::Ordering::Less => return (false, false),
```

Comment:
is the second bool equal to `false` ever an acceptable case? I.e. if everyone  restarts together and is waiting for their vote to land in a root, nobody will vote, and so everyone will stall right?

Reply:
`did_wait` can be false, yes. Because nodes could still have --wait-for-supermajority even when the network has passed the wait slot given and they are trying to join the network. They wouldn't wait at all because the slot they loaded from is already past the wait-for slot and the shred version matches. In this case they should wait to vote because they have joined the network that is already started producing blocks.